src/Base/Tools.h

```diff
@@ -113,31 +113,53 @@ inline manipulator<int> blanks(int n)
 // ----------------------------------------------------------------------------
 
 template<class T>
+    requires std::is_arithmetic_v<T>
 inline T clamp(T num, T lower, T upper)
 {
     return std::max<T>(std::min<T>(upper, num), lower);
 }
 
-template<class T>
-inline T sgn(T t)
+/// Returns -1, 0 or 1 depending on if the value is negative, zero or positive
+template<typename T>
+constexpr std::enable_if_t<std::is_arithmetic_v<T> && std::is_signed_v<T>, T> sgn(T val)
 {
-    if (t == 0) {
-        return T(0);
-    }
+    int oneIfPositive = int(0 < val);
+    int oneIfNegative = int(val < 0);
+    return T(oneIfPositive - oneIfNegative);
+}
 
-    return (t > 0) ? T(1) : T(-1);
+/// Convert degrees to radians, allow deduction for floating point types
+template<std::floating_point T>
+constexpr T toRadians(T degrees)
+{
+    constexpr auto degToRad = std::numbers::pi_v<T> / T(180);
+    return degrees * degToRad;
 }
 
-template<class T>
-inline T toRadians(T d)
+/// Convert degrees to radians, allow **explicit-only** for any arithmetic type
+template<typename T>
+    requires(std::is_arithmetic_v<T> && !std::floating_point<T>)
+constexpr T toRadians(std::type_identity_t<T> degrees)
 {
-    return static_cast<T>((d * std::numbers::pi) / 180.0);
+    using ResultT = std::conditional_t<std::is_integral_v<T>, double, T>;

```

Comment:
Can (not sure if should) also be moved into template as `typename R = std::conditional_t<std::is_integral_v<T>, double, T>`

Reply:
Thanks for pointing it out, though in this case I think it might be better to keep it like it is to avoid confusion about what the template parameters represent and how they should be used.

It could be a later addition if we find it missing.